### PR TITLE
Add com.visualstudio.code-oss

### PR DESCRIPTION
**Pull request by gasinvein**
It's a new opensource VS Code build, remade without custom manifest generatior.

For details on what's changed from the current stale build, see flathub/com.visualstudio.code.oss#56

### CODE REVIEW

**Comment by gasinvein** [comment]
bot, build com.visualstudio.code-oss

File: com.visualstudio.code-oss.yml [hunk 1]
```diff
@@ -0,0 +1,124 @@
+id: com.visualstudio.code-oss
```
Comment (gasinvein) [inline com.visualstudio.code-oss.yml:1]: There was a suggestion to use ID `io.github.microsoft.VSCode`. Maybe it would be better?

**Comment by tim77** [comment]
Please don't bury this build, really. :) There was done a lot of work and really nice to have such close to vanilla build.

Also JFYI: i tried to ask Theia devs (somewhat VS Code fork) for official Flatpak build, but no luck yet unfortunately. Maybe someone want continue to discuss: GitHub/theia-ide/theia-apps/343.

File: flathub.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+    "only-arches": [
+        "x86_64",
+        "aarch64"
+    ],
+    "publish-delay-hours": 1
```
Comment (barthalion) [inline flathub.json:6]: Please keep it at default.

File: install-data.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+import datetime
+from xml.dom import minidom
+
+FLATPAK_ID = os.environ.get('FLATPAK_ID')
+
+EXTRA_DESCRIPTION = [
+"""This is the Open Source build of Visual Studio Code, packaged into a Flatpak. Some features are
+different from the proprietary version: There is no telemetry nor Twitter integration, and the
+logo is a different one without copyright issue. This OSS repackaging, as well as the proprietary
+repackaging in Flathub, are not supported by Microsoft.""",
```
Comment (barthalion) [inline install-data.py:15]: What's the point of mentioning extra-data version here?

File: code-oss.sh [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    msg "Adding $tool_pythondir to PYTHONPATH"
+    if [ -z "$PYTHONPATH" ]; then
+      export PYTHONPATH="$tool_pythondir"
+    else
+      export PYTHONPATH="$PYTHONPATH:$tool_pythondir"
+    fi
+  fi
+done
+
+if [ "$FLATPAK_ENABLE_SDK_EXT" = "*" ]; then
```
Comment (barthalion) [inline code-oss.sh:44]: I'd change it to always load all available SDK extensions. That's more in line with users' expectations that Flatpak should magically pick up what has been installed.

File: install-data.py (hunk 3 above)
Comment (gasinvein) [reply install-data.py:15]: It's copied from current OSS build as-is, so dunno.

File: code-oss.sh (hunk 4 above)
Comment (gasinvein) [reply code-oss.sh:44]: It's the same behavior as of proprietary VS Code. Enabling all extensions unconditionally (at least via `enable.sh`) may result in conflicts, e.g. `node10` gets first on `PATH`, but `npm_config_nodedir` points to `node12` when both are installed.

Comment (barthalion) [reply code-oss.sh:44]: Fair enough.

File: install-data.py (hunk 3 above)
Comment (barthalion) [reply install-data.py:15]: I'd change it to something along "This build is not supported by Microsoft. Issues should be reported at https://github.com/flathub/appid." and drop the setence with link to https://github.com/flathub/com.visualstudio.code.oss/issues/6#issuecomment-380152999.

**Comment by gasinvein** [comment]
bot, build com.visualstudio.code-oss

**Pull request closed without merging**